# Import Packages

In [2]:
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
%matplotlib inline


In [3]:
#Read csv file
walk_run_data = pd.read_csv('run_walk.csv')

In [4]:
#check if file was loaded correctly
walk_run_data.head()

date                time username  wrist  activity  acceleration_x  \
0  2017-6-30  13:51:15:847724020   viktor      0         0          0.2650   
1  2017-6-30  13:51:16:246945023   viktor      0         0          0.6722   
2  2017-6-30  13:51:16:446233987   viktor      0         0          0.4399   
3  2017-6-30  13:51:16:646117985   viktor      0         0          0.3031   
4  2017-6-30  13:51:16:846738994   viktor      0         0          0.4814   

   acceleration_y  acceleration_z  gyro_x  gyro_y  gyro_z  
0         -0.7814         -0.0076 -0.0590  0.0325 -2.9296  
1         -1.1233         -0.2344 -0.1757  0.0208  0.1269  
2         -1.4817          0.0722 -0.9105  0.1063 -2.4367  
3         -0.8125          0.0888  0.1199 -0.4099 -2.9336  
4         -0.9312          0.0359  0.0527  0.4379  2.4922

In [5]:
walk_run_data.tail()

date               time username  wrist  activity  acceleration_x  \
88583  2017-7-9  20:9:15:317911028   viktor      0         0          0.3084   
88584  2017-7-9  20:9:15:517889022   viktor      0         0          0.4977   
88585  2017-7-9  20:9:15:717828989   viktor      0         0          0.4587   
88586  2017-7-9  20:9:15:917932987   viktor      0         0          0.2590   
88587  2017-7-9  20:9:16:117410004   viktor      0         0          0.3140   

       acceleration_y  acceleration_z  gyro_x  gyro_y  gyro_z  
88583         -0.8376         -0.1327  0.4823  2.0124  0.6048  
88584         -1.0027         -0.4397  0.1022 -1.2565 -0.0761  
88585         -1.1780         -0.2827 -1.4500 -0.2792 -1.2616  
88586         -0.8582         -0.0759 -1.5165  0.4560 -1.7755  
88587         -0.8008         -0.0911  0.1183  1.0850  1.2814

In [15]:
#check data types and data length
walk_run_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88588 entries, 0 to 88587
Data columns (total 11 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   date            88588 non-null  object 
 1   time            88588 non-null  object 
 2   username        88588 non-null  object 
 3   wrist           88588 non-null  int64  
 4   activity        88588 non-null  int64  
 5   acceleration_x  88588 non-null  float64
 6   acceleration_y  88588 non-null  float64
 7   acceleration_z  88588 non-null  float64
 8   gyro_x          88588 non-null  float64
 9   gyro_y          88588 non-null  float64
 10  gyro_z          88588 non-null  float64
dtypes: float64(6), int64(2), object(3)
memory usage: 7.4+ MB


In [16]:
#check data summary
walk_run_data.describe()

wrist      activity  acceleration_x  acceleration_y  \
count  88588.000000  88588.000000    88588.000000    88588.000000   
mean       0.522170      0.500801       -0.074811       -0.562585   
std        0.499511      0.500002        1.009299        0.658458   
min        0.000000      0.000000       -5.350500       -3.299000   
25%        0.000000      0.000000       -0.381800       -1.033500   
50%        1.000000      1.000000       -0.059500       -0.759100   
75%        1.000000      1.000000        0.355500       -0.241775   
max        1.000000      1.000000        5.603300        2.668000   

       acceleration_z        gyro_x        gyro_y        gyro_z  
count    88588.000000  88588.000000  88588.000000  88588.000000  
mean        -0.313956      0.004160      0.037203      0.022327  
std          0.486815      1.253423      1.198725      1.914423  
min         -3.753800     -4.430600     -7.464700     -9.480000  
25%         -0.376000     -0.920700     -0.644825     -1.345125  
50%         -0.221000      0.018700      0.039300      0.006900  
75%         -0.085900      0.888800      0.733700      1.398200  
max          1.640300      4.874200      8.498000     11.266200

# Data Cleaning

In [6]:
#check for black columns
walk_run_data.apply(lambda x: sum(x.isnull()),axis=0)
# =>our data is pretty clean

date              0
time              0
username          0
wrist             0
activity          0
acceleration_x    0
acceleration_y    0
acceleration_z    0
gyro_x            0
gyro_y            0
gyro_z            0
dtype: int64

In [9]:
#x values
features = ['acceleration_x','acceleration_y','acceleration_z','gyro_x','gyro_y','gyro_z'] 
X = walk_run_data[features]

#Y values
y= walk_run_data['activity']

# Model Planning

In [10]:
# split train and test data
from sklearn.model_selection import train_test_split

train_X,test_X,train_y,test_y = train_test_split(X,y,test_size = 0.2 , random_state = 1)

In [11]:
#check traing data shape
print(train_X.shape)
print(train_y.shape)

(70870, 6)
(70870,)


In [12]:
#check test data shape
print(test_X.shape)
print(test_y.shape)

(17718, 6)
(17718,)


# Model Building using Naive Bayes Algorithm

In [18]:
#import naive bayes algorithm
from sklearn.naive_bayes import GaussianNB

classifier = GaussianNB()

In [19]:
#train model
classifier.fit(train_X,train_y)

GaussianNB(priors=None, var_smoothing=1e-09)

In [20]:
#predict using the test data
y_predict = classifier.predict(test_X)

In [22]:
#check model accuracy
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_predict,test_y)
print(accuracy)

0.9554690145614629


In [24]:
# create a confusion matrix and classification report
from sklearn.metrics import confusion_matrix

conf_mat = confusion_matrix(y_predict,test_y)
print(conf_mat) 

# Calculating the accuracy from confusin matrix
# 1 - (90 + 699)/(8583 + 8346) = 0.95... => You get same value as our accuracy code

[[8583  699]
 [  90 8346]]


In [25]:
#classification report
from sklearn.metrics import classification_report

labels = ['Walk','Run']

print(classification_report(test_y,y_predict,target_names=labels))

              precision    recall  f1-score   support

        Walk       0.92      0.99      0.96      8673
         Run       0.99      0.92      0.95      9045

    accuracy                           0.96     17718
   macro avg       0.96      0.96      0.96     17718
weighted avg       0.96      0.96      0.96     17718



# Our Model is Really Good with an accuracy of 95 to 96 %